### ISSUE TITLE
MaxPoolingOp only supports NHWC. ERROR in my benchmark we would use NCHW dataformat

### ISSUE THREAD

**Issue by Leslie-Fang** [tensorflow/tensorflow#13664]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1. It must be a bug or a feature request.
2. The form below must be filled out.
3. It shouldn't be a TensorBoard issue. Those go [here](https://github.com/tensorflow/tensorboard/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 17.04
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: 1.4.0-rc0
- **Python version**: 2.7.13
- **Bazel version (if compiling from source)**: 0.6.1
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:
**numactl -m 1 python ./tensorflow/models/image/convnet-benchmark-alexnet/benchmark_alexnet_MKL.py --batch_size 256 --num_batches 100 -forward_backward_only --cpu knl 2>&1|tee tensorflow_alexnet_mkl.txt**

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

I have compiled tensorflow with bazel 0.6.1 and install it on my xeon phi TM platform.
Meanwhile I have got a benchmark related with alexnet.

When I run the benchmark I got the error:
_2017-10-12 09:38:14.503463: E tensorflow/core/common_runtime/executor.cc:643] Executor failed to create kernel. Invalid argument: Default MaxPoolingOp only supports NHWC.
         [[Node: pool1 = MaxPool[T=DT_FLOAT, data_format="NCHW", ksize=[1, 1, 3, 3], padding="VALID", strides=[1, 1, 2, 2], _device="/job:localhost/replica:0/task:0/device:CPU:0"](conv1)]]_

Also I find that in the benchmark, it has take both **"NCHW"** and **"NHWC"** data format into consideration. So the benchmark should support both data format.
I couldn't figure out why,  is this error relates with out compile process, or might I need to add some other option into the compile process.
Here is my compile command:

_bazel build --config=mkl --copt="-g" --copt="-DEIGEN_USE_VML" --copt="-mavx2" --copt="-mfma" --copt="-O3" --verbose_failures  --copt="-L/opt/intel/gcc/lib64" -s -c opt //tensorflow/tools/pip_package:build_pip_package_

anyone could help!

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by Leslie-Fang** [comment]
I want to change Default MaxPoolingOp only supports NHWC into NCHW. Is it possible?

**Comment by krak3nnn** [comment]
I'm having the same issue with TF 1.3 and 1.4. This seems to be an issue with MKL linking during "./configure" since MKL is what is responsible for interpreting the NCHW data format. 

I've noticed that in TF1.3 and 1.4, the configure no longer asks if I want to use MKL and if I do, if I wanted to download it automatically or use a path the MLKDNN. 

The last version that worked for me was TF 1.2.1 so I would recommend rebuilding or trying this [whl](https://anaconda.org/intel/tensorflow/1.2.1/download/tensorflow-1.2.1-cp27-cp27mu-linux_x86_64.whl).

**Comment by tatatodd** [comment]
@andydavis1 might have some clues about MKL related issues

**Comment by Leslie-Fang** [comment]
Hi, krak3nnn, thanks for your help. 
Today, I have got a compiled version of tensorflow1.3, which could run the benchmark with NHWC data format. So I think during the "./configure" step, could we add some options to solve this probem?

**Comment by andydavis1** [comment]
@vivek-rane Can you comment on the data layout for MKL Max Pooling op here?

**Comment by mahmoud-abuzaina** [comment]
This error was fixed in master branch, see #13562 . We are trying to get this fix into r1.4.

**Comment by tatatodd** [comment]
Thanks for the info @mahmoud-abuzaina!

In the meantime, the workarounds are to:
1. Build from sources including #13562, or
2. Use a nightly that includes this fix, or
3. Use r1.4rc1 when it is available, or the actual r1.4 release when it is available.

**Comment by Leslie-Fang** [comment]
Thanks a lot for your information @mahmoud-abuzaina @tatatodd , I would try the new version.

**Comment by Leslie-Fang** [comment]
Hi, @mahmoud-abuzaina @tatatodd  I have try build the TF in the master branch:commit 7cdd26f606b39e3e487ec15dfa6eb5c6cf63ef84 with bazel 0.6.1
It failed with the error:
_ERROR: /root/tensorflow/tensorflow/python/BUILD:1373:1: Linking of rule '//tensorflow/python:gen_string_ops_py_wrappers_cc' failed (Exit 1): gcc failed: error executing command
bazel-out/host/bin/_solib_k8/_U_S_Stensorflow_Spython_Cgen_Ustring_Uops_Upy_Uwrappers_Ucc___Utensorflow/libtensorflow_framework.so: error: undefined reference to 'tensorflow::MklCPUAllocator::kName'
bazel-out/host/bin/_solib_k8/_U_S_Stensorflow_Spython_Cgen_Ustring_Uops_Upy_Uwrappers_Ucc___Utensorflow/libtensorflow_framework.so: error: undefined reference to 'tensorflow::MklCPUAllocator::kMaxLimitStr'_

Do you have any idea?

**Comment by mahmoud-abuzaina** [comment]
@Leslie-Fang this is a more recent issue and should be fixed when #13697 is merged.

**Comment by brave-hannah** [comment]
I meet this issue also with tf-1.8 when I do distributed train via benchmark, even while I have set num_gpus to 2 but it seems using cpu both nodes without control, which makes me confused at all...

> I0711 13:18:38.419439 140188219279168 tf_logging.py:116] Running local_init_op.
I0711 13:18:41.063343 140188219279168 tf_logging.py:116] Done running local_init_op.
Running warm up
2018-07-11 13:18:45.151136: E tensorflow/core/common_runtime/executor.cc:660] Executor failed to create kernel. Invalid argument: Default MaxPoolingOp only supports NHWC on device type CPU
         [[Node: v0/tower_0/cg/mpool0/MaxPool = MaxPool[T=DT_FLOAT, data_format="NCHW", ksize=[1, 1, 3, 3], padding="SAME", strides=[1, 1, 2, 2], _device="/job:worker/replica:0/task:1/device:CPU:0"](v0/tower_0/cg/conv0/Relu)]]
I0711 13:18:45.186572 140188219279168 tf_logging.py:116] Error reported to Coordinator: <class 'tensorflow.python.framework.errors_impl.InvalidArgumentError'>, Default MaxPoolingOp only supports NHWC on device type CPU
         [[Node: v0/tower_0/cg/mpool0/MaxPool = MaxPool[T=DT_FLOAT, data_format="NCHW", ksize=[1, 1, 3, 3], padding="SAME", strides=[1, 1, 2, 2], _device="/job:worker/replica:0/task:1/device:CPU:0"](v0/tower_0/cg/conv0/Relu)]]

**Comment by sherif-med** [comment]
This error maybe due to inappropriately installation of cuda. tensorflow-gpu have no access to GPU devices. Try to reinstall cuda and include cudnn and check if tensorflow can use GPUs by running:

`sess = tf.Session(config=tf.ConfigProto(log_device_placement=True))`
